app/controllers/articles_controller.rb

```diff
@@ -89,6 +91,7 @@ def redirect_url(article)
 
   def set_wip
     @article.wip = params[:commit] == 'WIP'
+    @article.generate_token_with_secure_random_base64
```

Comment:
```suggestion
    @article.generate_token! if @aritcle.wip
```

Articleの方のコメントと合わせて、こんな感じにした方が実装の外側から見た人にとってはわかりやすいかなとおもいます。

オブジェクト指向プログラミング的には、メソッドシグネチャーだけをみて動作が予想できるのが理想なので。